### ISSUE TITLE
Uploads fail with 422 error with Git LFS 2.5.0

### ISSUE THREAD

**Issue by zjslqshqz** [git-lfs/git-lfs#3158]
Software version
```
git-lfs/2.5.0 (GitHub; darwin 386; go 1.10.3; git 93775601)
gitlab 11.2
```
```
Locking support detected on remote "origin". Consider enabling it with:
  $ git config lfs.https://gitlab.gitlab.com/t.git/info/lfs.locksverify true
Uploading LFS objects:   0% (0/1), 0 B | 0 B/s                                                               
LFS: Client error: https://gitlab.gitlab.com/t.git/gitlab-lfs/objects/80cf36a3a93314dcae215ae8918b8b0faef1c5fb2b1af917ab3632edf5628e43/26 from HTTP 422
Uploading LFS objects:   0% (0/1), 0 B | 0 B/s, done
error: failed to push some refs to 'https://gitlab.gitlab.com/t.git'

```
BUT!

Downgrading to git-lfs=2.4.1 fixed the issue for me.

Is probably on to something with the Content-Type header theory going from:
Content-Type: application/octet-stream in 2.4.1
to the more appropiate
Content-Type: application/zip in 2.5.0.

So this problem can only be solved by downgrade?

**Comment by ttaylorr** [comment]
Hi @zjslqshqz, thank you for letting us know, and we're sorry about the regression.

I think that I could have handled merging the change that introduced this behavior in a different way that would have alerted other hosting platforms, so I think that we have a few options.

1. We should (first and foremost) alert GitLab (as well as any other hosting platforms that are broken by this change) about what has happened and what they can do to go forward.

2. We should either (A) introduce an option (a-la `lfs.[url].locksverify`) to disable content-type detection in a host-specific configuration, or (B) disable the feature entirely in v2.5.1 while we wait for (1) to take effect.

I would like to get the consensus of other core team members, incl. @PastelMobileSuit and @larsxschneider. Let's figure out what to do, and then do so.

**Comment by zjslqshqz** [comment]
@ttaylorr Thank you for your reply, I understand

**Comment by PastelMobileSuit** [comment]
@ttaylorr I'm in favor of 2a, since we've already done the work to add the feature and it does work on some hosting platforms. My opinion is that we should default this option to `false` and have the feature be opt-in until other hosting platforms support it.

**Comment by ttaylorr** [comment]
> @ttaylorr I'm in favor of 2a, since we've already done the work to add
> the feature and it does work on some hosting platforms.

Sounds good, and thanks for weighing in. I implemented just that in the
pull request that is referenced here.

I think that this allows us to keep the feature on for the majority of
hosts that support it, while allowing us to scope it in and turn it off
for those that do not. That seems like a good balance to me.

I also wired through into the transfer queue that we notify users of a
'git config' command that they can run iff we encounter an HTTP 422 when
uploading that should alleviate the 'Content-Type' troubles.
